securedrop_client/gui/widgets.py

```diff
@@ -2751,11 +2754,23 @@ def update_conversation(self, collection: list) -> None:
         # items corresponding to deleted items in the source collection.
         current_conversation = self.current_messages.copy()
 
+        # To hold widgets that are in a failed or pending state which need to
+        # be appended to the end of the conversation until their state is
+        # updated from the server.
+        pending_items = []
+
         for index, conversation_item in enumerate(collection):
             item_widget = current_conversation.get(conversation_item.uuid)
             if item_widget:
                 current_conversation.pop(conversation_item.uuid)
-                if item_widget.index != index:
+                if isinstance(item_widget, ReplyWidget) and item_widget.reply_status != "SUCCEEDED":
```

Comment:
i think this if-elif block doesn't cover the case for reply widgets that are successful where `item_widget.index == index`